bug: unreachable 'error' branch — case 'warning' || 'error' in individual.tsx

Found while triaging Renovate PR #44 (TypeScript 5.9.3), which fails typecheck with:

```
components/results/individual.tsx(95,12): error TS2872: This kind of expression is always truthy.
```

The code is:
```ts
case 'warning' || 'error':
```

`'warning' || 'error'` evaluates to `'warning'` — the **`'error'` branch is unreachable today on main**. This is a live logic bug (error-level results fall through to whatever the default handling is), invisible until TS 5.8 added the TS2872 diagnostic.

**Fix:** split into fall-through clauses:
```ts
case 'warning':
case 'error':
```

**Unblocks:** Renovate #44 (TS 5.9.3, a minor that will then automerge) and is a prerequisite for #78 (TS 6).

--- a/hyperglass/ui/components/results/individual.tsx
+++ b/hyperglass/ui/components/results/individual.tsx
@@ -30,6 +30,7 @@ import {
   useTableToString,
 } from '~/hooks';
 import { isStringOutput, isStructuredOutput } from '~/types';
+import { levelToStatus } from '~/util';
 import { CopyButton } from './copy-button';
 import { FormattedError } from './formatted-error';
 import { isFetchError, isLGError, isLGOutputOrError, isStackError } from './guards';
@@ -87,16 +88,7 @@ const _Result: React.ForwardRefRenderFunction<HTMLDivElement, ResultProps> = (
   const [lastResponseAt, setLastResponseAt] = useState<number>(() => Date.now());
 
   const setErrorLevel = (level: ResponseLevel): void => {
-    let e: ErrorLevels = 'error';
-    switch (level) {
-      case 'success':
-        e = level;
-        break;
-      case 'warning' || 'error':
-        e = 'warning';
-        break;
-    }
-    _setErrorLevel(e);
+    _setErrorLevel(levelToStatus(level));
   };
 
   // When snapshot is provided, coerce it to a QueryResponse shape so all

--- a/hyperglass/ui/util/common.test.ts
+++ b/hyperglass/ui/util/common.test.ts
@@ -1,5 +1,28 @@
 import { describe, expect, it, test } from 'vitest';
-import { all, andJoin, chunkArray, dedupObjectArray, entries, isFQDN } from './common';
+import {
+  all,
+  andJoin,
+  chunkArray,
+  dedupObjectArray,
+  entries,
+  isFQDN,
+  levelToStatus,
+} from './common';
+
+describe('levelToStatus - map backend response level to alert status', () => {
+  it('passes success through', () => {
+    expect(levelToStatus('success')).toBe('success');
+  });
+  it('passes warning through', () => {
+    expect(levelToStatus('warning')).toBe('warning');
+  });
+  it('maps error to error (regression: was unreachable via `warning || error` case)', () => {
+    expect(levelToStatus('error')).toBe('error');
+  });
+  it("maps danger to error (no 'danger' alert status exists)", () => {
+    expect(levelToStatus('danger')).toBe('error');
+  });
+});
 
 test('all - all items are truthy', () => {
   // biome-ignore lint/suspicious/noSelfCompare: because this is a test, duh

--- a/hyperglass/ui/util/common.ts
+++ b/hyperglass/ui/util/common.ts
@@ -1,3 +1,22 @@
+import type { ErrorLevels } from '~/types';
+
+/**
+ * Map a backend response level onto an alert status.
+ *
+ * `ResponseLevel` includes `danger`, which has no alert-status equivalent and
+ * renders as `error`. (Regression guard: an earlier `case 'warning' || 'error'`
+ * expression made the `error` branch unreachable — see issue #106.)
+ */
+export function levelToStatus(level: ResponseLevel): ErrorLevels {
+  switch (level) {
+    case 'success':
+    case 'warning':
+      return level;
+    default:
+      return 'error';
+  }
+}
+
 export function all<I extends unknown>(...iter: I[]): boolean {
   for (const i of iter) {
     if (!i) {

